src/tools/rustfmt/src/expr.rs

```diff
@@ -682,9 +682,20 @@ fn to_control_flow(expr: &ast::Expr, expr_type: ExprType) -> Option<ControlFlow<
                 expr.span,
             ))
         }
-        ast::ExprKind::ForLoop(ref pat, ref cond, ref block, label) => {
-            Some(ControlFlow::new_for(pat, cond, block, label, expr.span))
-        }
+        ast::ExprKind::ForLoop {
+            ref pat,
+            ref iter,
+            ref body,
+            label,
+            kind,
+        } => Some(ControlFlow::new_for(
+            pat,
+            iter,
+            body,
+            label,
+            expr.span,
+            kind == ast::ForLoopKind::ForAwait,
```

Comment:
Just pass through the `ForLoopKind`, this seems just as complicated but easier to get wrong 😆